Mathlib/Topology/UrysohnsLemma.lean

```diff
@@ -398,3 +400,68 @@ theorem exists_continuous_one_zero_of_isCompact [RegularSpace X] [LocallyCompact
     exact interior_subset hx
   · have : 0 ≤ f x ∧ f x ≤ 1 := by simpa using h'f x
     simp [this]
+
+/-- Urysohn's lemma: if `s` and `t` are two disjoint sets in a regular locally compact topological
+space `X`, with `s` compact and `t` closed, then there exists a continuous compactly supported
+function `f : X → ℝ` such that
+
+* `f` equals one on `s`;
+* `f` equals zero on `t`;
+* `0 ≤ f x ≤ 1` for all `x`.
+
+Moreover, if `s` is Gδ, one can ensure that `f ⁻¹ {1}` is exactly `s`.
+-/
+theorem exists_continuous_one_zero_of_isCompact_of_isGδ [RegularSpace X] [LocallyCompactSpace X]
+    {s t : Set X} (hs : IsCompact s) (h's : IsGδ s) (ht : IsClosed t) (hd : Disjoint s t) :
+    ∃ f : C(X, ℝ), s = f ⁻¹' {1} ∧ EqOn f 0 t ∧ HasCompactSupport f
+      ∧ ∀ x, f x ∈ Icc (0 : ℝ) 1 := by
+  rcases isGδ_iff_eq_iInter_nat.1 h's with ⟨U, U_open, hU⟩
```

Comment:
Nit: could you please add a dot-notation `alias` for the `.mp` part of this lemma and use it as `h's.eq_iInter_nat` here?
Otherwise LGTM. Thanks!
bors d+